avro-serializer/src/main/java/io/confluent/kafka/serializers/AbstractKafkaAvroSerializer.java

```diff
@@ -76,4 +68,7 @@
       throw new SerializationException("Error serializing Avro message", e);
     }
   }
+
+  protected abstract DatumWriter<Object> getDatumWriter(Schema schema, Object object);
```

Comment:
Any reason the logic isn't shared by all the implementations here rather than being duplicated across all the implementations? I can see that they differ in that only one implements the class lookup functionality, but I'm unclear why we wouldn't want the same setting to be available across all subclasses of this class.